src/main/scala/uk/co/gresearch/spark/diff/Diff.scala

```diff
@@ -144,20 +145,20 @@ class Differ(options: DiffOptions) {
 
   private def getChangeColumn(
       existsColumnName: String,
-      valueColumns: Seq[String],
+      valueVolumnsWithComparator: Seq[(String, DiffComparator)],
       left: Dataset[_],
       right: Dataset[_]
   ): Option[Column] = {
     options.changeColumn
       .map(changeColumn =>
         when(left(existsColumnName).isNull || right(existsColumnName).isNull, lit(null))
           .otherwise(
-            Some(valueColumns.toSeq)
+            Some(valueVolumnsWithComparator)
               .filter(_.nonEmpty)
               .map(columns =>
                 concat(
                   columns
-                    .map(c => when(left(backticks(c)) <=> right(backticks(c)), array()).otherwise(array(lit(c)))): _*
+                    .map(entry => when(entry._2.equiv(left(backticks(entry._1)), right(backticks(entry._1))), array()).otherwise(array(lit(entry._1)))): _*
```

Comment:
I'd prefer naming the tuple elements, rather than using `_1` and `_2`:
```suggestion
                    .map { case (c, cmp) =>
                      when(cmp.equiv(left(backticks(c)), right(backticks(c))), array()).otherwise(array(lit(c)))
                    }: _*
```